rust-client/src/main.rs

```diff
@@ -204,8 +200,18 @@ async fn init_swarm(
     )
     .upgrade(upgrade::Version::V1)
     .authenticate(noise::NoiseConfig::xx(noise_keys).into_authenticated())
-    .multiplex(libp2p::yamux::YamuxConfig::default())
-    .boxed();
+    .multiplex(libp2p::yamux::YamuxConfig::default());
+
+    let mut quic_config = quic::Config::new(&local_key);
+    quic_config.support_draft_29 = true;
+    let quic_transport = quic::async_std::Transport::new(quic_config);
+
+    let transport = OrTransport::new(quic_transport, tcp_relay_transport)
+        .map(|either_output, _| match either_output {
+            EitherOutput::First((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+            EitherOutput::Second((peer_id, muxer)) => (peer_id, StreamMuxerBox::new(muxer)),
+        })
+        .boxed();
```

Comment:
This would not allow a DNS address for QUIC, right? E.g. `/dns6/max-inden.de/udp/1234/quic`.

I think the easiest fix would be to wrap the `quic_transport` with another `DnsConfig::system`.

Reply:
Done in 780e104.